# Kidney Stones notebook

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
train = pd.read_csv('train.csv')
test = pd.read_csv('test.csv')

In [4]:
train.head()

id  gravity    ph  osmo  cond  urea  calc  target
0   0    1.013  6.19   443  14.8   124  1.45       0
1   1    1.025  5.40   703  23.6   394  4.18       0
2   2    1.009  6.13   371  24.5   159  9.04       0
3   3    1.021  4.91   442  20.8   398  6.63       1
4   4    1.021  5.53   874  17.8   385  2.21       1

In [5]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 414 entries, 0 to 413
Data columns (total 8 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   id       414 non-null    int64  
 1   gravity  414 non-null    float64
 2   ph       414 non-null    float64
 3   osmo     414 non-null    int64  
 4   cond     414 non-null    float64
 5   urea     414 non-null    int64  
 6   calc     414 non-null    float64
 7   target   414 non-null    int64  
dtypes: float64(4), int64(4)
memory usage: 26.0 KB


In [6]:
train.describe()

id     gravity          ph         osmo        cond  \
count  414.000000  414.000000  414.000000   414.000000  414.000000   
mean   206.500000    1.017894    5.955459   651.545894   21.437923   
std    119.655756    0.006675    0.642260   234.676567    7.514750   
min      0.000000    1.005000    4.760000   187.000000    5.100000   
25%    103.250000    1.012000    5.530000   455.250000   15.500000   
50%    206.500000    1.018000    5.740000   679.500000   22.200000   
75%    309.750000    1.022000    6.280000   840.000000   27.600000   
max    413.000000    1.040000    7.940000  1236.000000   38.000000   

             urea        calc      target  
count  414.000000  414.000000  414.000000  
mean   278.657005    4.114638    0.444444  
std    136.442249    3.217641    0.497505  
min     10.000000    0.170000    0.000000  
25%    170.000000    1.450000    0.000000  
50%    277.000000    3.130000    0.000000  
75%    385.000000    6.630000    1.000000  
max    620.000000   14.340000    1.000000

#### Splitting the datasets for training and testing

In [7]:
X_train = train.drop('target',axis=1)
y_train = train['target']
X_test = test

In [8]:
test_id = test['id']

Drop the id column from both training and testing dataset as it is not required while fitting and predicting.

In [9]:
X_train = X_train.drop('id',axis=1)
X_test = X_test.drop('id',axis=1)

## Logistic Regression Model

In [10]:
from sklearn.linear_model import LogisticRegression
model = LogisticRegression()
model.fit(X_train,y_train)

LogisticRegression()

In [11]:
model.predict(X_test)
Accuracy = model.score(X_train,y_train)*100
Accuracy

72.46376811594203

## Random Forest Classifier Model

In [12]:
from sklearn.ensemble import RandomForestClassifier
RFC = RandomForestClassifier()
RFC.fit(X_train, y_train)

RandomForestClassifier()

In [13]:
RFC.predict(X_test)
Rand_Accuracy = RFC.score(X_train,y_train)*100
Rand_Accuracy

100.0

## Decision Tree Classifer Model

In [14]:
from sklearn.tree import DecisionTreeClassifier
DTC = DecisionTreeClassifier()
DTC.fit(X_train, y_train)

DecisionTreeClassifier()

In [15]:
DTC.predict(X_test)
Decision_Accuracy = DTC.score(X_train,y_train)*100
Decision_Accuracy

100.0

Both the Decision Tree Classifier and the Random Forest Classifier model outperformed the Logistic Regression model each attaining a model accuracy of 100%. Logistic Regression model had an accuracy of 72.46%.

## Sample Submission

In [16]:
Id = pd.DataFrame(test_id, columns=['id'])
Predictions = pd.DataFrame(RFC.predict(X_test), columns=['Predicted'])
Rand_Predictions  = pd.concat([Id, Predictions], axis=1)

In [17]:
Predictions.value_counts()

Predicted
0            153
1            123
dtype: int64

In [18]:
Predictions1 = pd.DataFrame(DTC.predict(X_test), columns=['Predicted'])
Decision_Predictions = pd.concat([Id, Predictions1], axis=1)

In [19]:
Predictions1.value_counts()

Predicted
0            153
1            123
dtype: int64

In [21]:
Rand_Predictions.to_csv('Kidney Stones.csv', index=False)
Decision_Predictions.to_csv('Kidney_Stones.csv', index=False)